src/main/java/org/jenkinsci/test/acceptance/junit/Native.java

```diff
@@ -45,9 +63,34 @@ public void evaluate() throws Throwable {
 
                 private void verifyNativeCommandPresent(Native n) throws IOException, InterruptedException {
                     if (n==null)        return;
+                    
+                    // Checks performed in this annotation only makes sense if Jenkins and Tests are being executed in the same machine
+                    if (!(controller instanceof LocalController)) {
+                        throw new AssumptionViolatedException("Test skipped. Native should be used with a local controller, otherwise it cannot be fully trusted to work as expected.");
```

Comment:
> before it could do this by virtue of creating a dummy file in my path

This seems like a use case we need not support. If you care, fix `Native` to actually query the remote system.